lyaemu/coarse_grid.py

```diff
@@ -272,6 +282,71 @@ def get_emulator(self, max_z=4.2):
         gp = self._get_custom_emulator(emuobj=None, max_z=max_z)
         return gp
 
+    def get_pdf_vectors(self, max_z=4.2, kfunits="kms"):
+        """Get the desired flux vectors and their parameters"""
+        pvals = self.get_parameters()
+        nparams = np.shape(pvals)[1]
+        nsims = np.shape(pvals)[0]
+        assert nparams == len(self.param_names)
+        myspec = deltaF_pdf.MySpectra(max_z=max_z, bins=self.bins)
+        aparams = pvals
+        #Note this gets tau_0 as a linear scale factor from the observed power law
+        dpvals = self.mf.get_params()
+        nuggets = np.zeros_like(pvals[:,0])
+        #Savefile prefix
+        mfc = "cc"
+        if dpvals is not None:
+            #Add a small offset to the mean flux in each simulation to improve support
+            nuggets = np.arange(nsims)/nsims * (dpvals[-1] - dpvals[0])/(np.size(dpvals)+1)
+            newdp = dpvals[0] + (dpvals-dpvals[0]) / (np.size(dpvals)+1) * np.size(dpvals)
+            #Make sure we don't overflow the parameter limits
+            assert (newdp[-1] + nuggets[-1] < dpvals[-1]) and (newdp[0] + nuggets[0] >= dpvals[0])
+            dpvals = newdp
+            aparams = np.array([np.concatenate([dp+nuggets[i],pvals[i]]) for dp in dpvals for i in range(nsims)])
+            mfc = "mf"
+        try:
+            deltaF_bins, pdf_vectors = self.load_pdf_vectors(aparams, mfc=mfc)
+        except (AssertionError, OSError):
+            print("Could not load flux vectors, regenerating from disc")
+            pdfs = [self._get_pv(pp, myspec) for pp in pvals]
+            mef = lambda pp: self.mf.get_mean_flux(myspec.zout, params=pp)[0]
+            if dpvals is not None:
+                deltaF_vectors = np.array([pdfs[i].get_deltaF_pdf(mean_fluxes = mef(dp+nuggets[i])) for dp in dpvals for i in range(nsims)])
+                deltaF_bins = pdfs[0].bins
+
+            else:
+                deltaF_vectors = np.array([pdfs[i].get_deltaF_pdf(mean_fluxes = mef(dpvals)) for i in range(nsims)])
+                deltaF_bins = pdfs[0].bins
+
+            #Same in all boxes
+            self.save_pdf_vectors(aparams, deltaF_bins, deltaF_vectors, mfc=mfc)
+        assert np.shape(deltaF_vectors)[0] == np.shape(aparams)[0]
+        return aparams, deltaF_bins, deltaF_vectors
```

Comment:
Is this just the same as the flux_vector function? In which case a better thing to do is to reuse that function with maybe slight changes to the default arguments.